lib/http2.c

```diff
@@ -1166,8 +1169,10 @@ void Curl_http2_done(struct connectdata *conn, bool premature)
 
   if(premature) {
     /* RST_STREAM */
-    nghttp2_submit_rst_stream(httpc->h2, NGHTTP2_FLAG_NONE, http->stream_id,
-                              NGHTTP2_STREAM_CLOSED);
+    if(!nghttp2_submit_rst_stream(httpc->h2, NGHTTP2_FLAG_NONE,
+                                  http->stream_id, NGHTTP2_STREAM_CLOSED))
+      (void)h2_session_send(data, httpc->h2);
```

Comment:
``h2_session_send()`` deals with stream priorities, but this stream has just been reset. Why not call ``nghttp2_session_send()`` directly?

Reply:
No biggie really, but I was thinking it is better to use a common path for sending data. But you're right, I can shortcut that just in case the prios are actually changed at that point...